packages/snippet-bot/src/snippet-bot.ts

```diff
@@ -12,28 +12,127 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import {Application} from 'probot';
+import {Application, Octokit} from 'probot';
+import {parseRegionTags} from './region-tag-parser';
 import {logger} from 'gcf-utils';
+import * as minimatch from 'minimatch';
+
+type Conclusion =
+  | 'success'
+  | 'failure'
+  | 'neutral'
+  | 'cancelled'
+  | 'timed_out'
+  | 'action_required'
+  | undefined;
+
+interface ConfigurationOptions {
+  ignoreFiles: string[];
+}
 
 const CONFIGURATION_FILE_PATH = 'snippet-bot.yml';
 
-interface Configuration {
-  randomBoolean: boolean;
+class Configuration {
+  private options: ConfigurationOptions;
+  private minimatches: minimatch.IMinimatch[];
+
+  constructor(options: ConfigurationOptions) {
+    this.options = options;
+    this.minimatches = (options as ConfigurationOptions).ignoreFiles.map(
+      pattern => {
+        return new minimatch.Minimatch(pattern);
+      }
+    );
+  }
+
+  ignoredFile(filename: string): boolean {
+    return this.minimatches.some(mm => {
+      return mm.match(filename);
+    });
+  }
 }
 
 export = (app: Application) => {
-  app.on(['issues.opened', 'pull_request.opened'], async context => {
-    const config = (await context.config(
+  app.on('pull_request', async context => {
+    const configOptions = (await context.config(
```

Comment:
It's important we wrap this call here in a `try`/`catch`.  See:
https://github.com/googleapis/repo-automation-bots/blob/master/packages/sync-repo-settings/src/sync-repo-settings.ts#L81

If the bot doesn't have code read access on setup, it will throw

Reply:
Done